### ISSUE TITLE
Installation issue: gettext ~libxml2

### ISSUE THREAD

**Issue by wdconinc** [LLNL/spack#46815]
### Steps to reproduce the issue

In github codespace on develop branch:
```console
$ spack install gettext ~libxml2
```
fails with
```
     7622    In file included from term-styled-ostream.oo.c:26:
  >> 7623    ./libcroco/cr-sel-eng.h:43:10: fatal error: libxml/tree.h: No such file or directory
     7624       43 | #include <libxml/tree.h>
     7625          |          ^~~~~~~~~~~~~~~
     7626    compilation terminated.
```

### Error message

<details><summary>Error message</summary>
<pre>
==> Installing gettext-0.22.5-2dnfdxzyp4qqpil5vm77j5vqmqnso5tu [14/14]
==> No binary for gettext-0.22.5-2dnfdxzyp4qqpil5vm77j5vqmqnso5tu found: installing from source
==> Fetching https://mirror.spack.io/_source-cache/archive/fe/fe10c37353213d78a5b83d48af231e005c4da84db5ce88037d88355938259640.tar.xz
==> Ran patch() for gettext
==> gettext: Executing phase: 'autoreconf'
==> gettext: Executing phase: 'configure'
==> gettext: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j2' 'V=1'

6 errors found in build log:
     7617    /workspaces/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H -I. -I..  -I. -I. -I.. -I.. -Iglib -DIN_LIBTEXTSTYLE     -I./libcroco  -DDEPENDS_ON_LIBICONV=1 -I/workspaces/spack/opt/spa
             ck/linux-ubuntu20.04-zen2/gcc-9.4.0/libiconv-1.17-djsbdrvvlfszikanwgo674xwecdqzhge/include  -g -O2 -w -c styled-ostream.c
     7618    /workspaces/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H -I. -I..  -I. -I. -I.. -I.. -Iglib -DIN_LIBTEXTSTYLE     -I./libcroco  -DDEPENDS_ON_LIBICONV=1 -I/workspaces/spack/opt/spa
             ck/linux-ubuntu20.04-zen2/gcc-9.4.0/libiconv-1.17-djsbdrvvlfszikanwgo674xwecdqzhge/include  -g -O2 -w -c sys_socket.c
     7619    /workspaces/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H -I. -I..  -I. -I. -I.. -I.. -Iglib -DIN_LIBTEXTSTYLE     -I./libcroco  -DDEPENDS_ON_LIBICONV=1 -I/workspaces/spack/opt/spa
             ck/linux-ubuntu20.04-zen2/gcc-9.4.0/libiconv-1.17-djsbdrvvlfszikanwgo674xwecdqzhge/include  -g -O2 -w -c term-ostream.c
     7620    /workspaces/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H -I. -I..  -I. -I. -I.. -I.. -Iglib -DIN_LIBTEXTSTYLE     -I./libcroco  -DDEPENDS_ON_LIBICONV=1 -I/workspaces/spack/opt/spa
             ck/linux-ubuntu20.04-zen2/gcc-9.4.0/libiconv-1.17-djsbdrvvlfszikanwgo674xwecdqzhge/include  -g -O2 -w -c term-style-control.c
     7621    /workspaces/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H -I. -I..  -I. -I. -I.. -I.. -Iglib -DIN_LIBTEXTSTYLE     -I./libcroco  -DDEPENDS_ON_LIBICONV=1 -I/workspaces/spack/opt/spa
             ck/linux-ubuntu20.04-zen2/gcc-9.4.0/libiconv-1.17-djsbdrvvlfszikanwgo674xwecdqzhge/include  -g -O2 -w -c term-styled-ostream.c
     7622    In file included from term-styled-ostream.oo.c:26:
  >> 7623    ./libcroco/cr-sel-eng.h:43:10: fatal error: libxml/tree.h: No such file or directory
     7624       43 | #include <libxml/tree.h>
     7625          |          ^~~~~~~~~~~~~~~
     7626    compilation terminated.
  >> 7627    make[4]: *** [Makefile:6192: config.h] Error 1
     7628    make[4]: Leaving directory '/tmp/codespace/spack-stage/spack-stage-gettext-0.22.5-2dnfdxzyp4qqpil5vm77j5vqmqnso5tu/spack-src/libtextstyle/lib'
  >> 7629    make[3]: *** [Makefile:1999: all-recursive] Error 1
     7630    make[3]: Leaving directory '/tmp/codespace/spack-stage/spack-stage-gettext-0.22.5-2dnfdxzyp4qqpil5vm77j5vqmqnso5tu/spack-src/libtextstyle'
  >> 7631    make[2]: *** [Makefile:1927: all] Error 2
     7632    make[2]: Leaving directory '/tmp/codespace/spack-stage/spack-stage-gettext-0.22.5-2dnfdxzyp4qqpil5vm77j5vqmqnso5tu/spack-src/libtextstyle'
  >> 7633    make[1]: *** [Makefile:408: all-recursive] Error 1
     7634    make[1]: Leaving directory '/tmp/codespace/spack-stage/spack-stage-gettext-0.22.5-2dnfdxzyp4qqpil5vm77j5vqmqnso5tu/spack-src'
  >> 7635    make: *** [Makefile:364: all] Error 2

See build log for details:
  /tmp/codespace/spack-stage/spack-stage-gettext-0.22.5-2dnfdxzyp4qqpil5vm77j5vqmqnso5tu/spack-build-out.txt
</pre></details>

### Information on your system

* **Spack:** 0.23.0.dev0 (dce3cf6f31de36f97e4e3d8e3fcc00c92f5a3d92)
* **Python:** 3.12.1
* **Platform:** linux-ubuntu20.04-zen2

### Additional information

[spack-build-env.txt](https://github.com/user-attachments/files/17267081/spack-build-env.txt)
[spack-build-out.txt](https://github.com/user-attachments/files/17267082/spack-build-out.txt)

This is potentially caused by the unvendoring of libxml2 in https://github.com/spack/spack/pull/43622 @haampie 

Maintainer tag: @michaelkuhn

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by yizeyi18** [comment]
This should be generated twice by `Makefile` of `libtextstyle` and `gettext-tools/gnulib-lib`, both of them have a target `libxml/tree.h`. Is the target not made or missing? Can you help checking `spack-src/libtextstyle/lib/Makefile` in which should have something like `libxml/tree.h: `?

**Comment by bernhardkaindl** [comment]
@wdconinc, @yizeyi18: This is fixed by #46870
